### PR TITLE
Feature/create target

### PR DESCRIPTION
**Pull request by manema**
### Main Title: Create target

Trello card:
https://trello.com/c/FGbCcDyj/4-as-a-user-i-should-be-able-to-create-a-new-target

Description:
- New create target form.
- Backend integration to create a new target.

_Important: This code's version doesn't show targets on the map, I'm gonna add this feature in the next PR, since the current one was getting too big, thus difficult to review._

create_target com-video-to-gif

### CODE REVIEW

File: src/actions/mapActions.js [hunk 1]
```diff
@@ -19,3 +32,22 @@ export const getPosition = () =>
       throw err;
     }
   };
+
+export const createTarget = ({ area, latitude, longitude, title, topic }) =>
+  async (dispatch) => {
+    try {
+      const target = {
```
Comment (felipegarcia92) [inline src/actions/mapActions.js]: I think you can rename the destructured parameters and just make this line:
```
export const createTarget = ({ area: radius, latitude: lat, longitude: lng, title, topic: topicId }) =>
.
.
.
const target = { title, lat, lng, radius, topicId}
```

File: src/containers/MainScreen/index.js [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
+  state = {
+    showMenu: false
+  }
+
   componentDidMount() {
     const { getPosition } = this.props;
     getPosition();
   }
 
+  onShowMenu = () => {
```
Comment (felipegarcia92) [inline src/containers/MainScreen/index.js]: Small detail, but "show menu" is not the action or event the user is doing. It's toggling (if it's on, it goes off, and viceversa). So maybe `toggleMenu` or `toggleShowMenu` is a more accurate name for this.

File: src/actions/mapActions.js [hunk 3]
```diff
@@ -33,21 +33,21 @@ export const getPosition = () =>
...
-    }
-  };
+export const createTarget =
+  ({
+    area: radius,
+    latitude: lat,
+    longitude: lng,
+    title,
+    topic: topicId }) =>
+    async (dispatch) => {
```
Comment (felipegarcia92) [inline src/actions/mapActions.js]: Check indentation here, curly brackets should be on the same level. Also, parameters are on the same level as the code-block.

```
({
  area: radius,
  latitude: lat,
  longitude: lng,
  title,
  topic: topicId 
}) =>
    async (dispatch) => {
       try {
          ...
        }
```

File: src/components/targets/CreateTargetForm/index.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+CreateTargetForm.propTypes = {
+  handleSubmit: func.isRequired,
+  submitting: bool.isRequired,
+  error: string
+};
+
+export default reduxForm({
+  form: 'createTarget',
+  validate: constraints.validations(constraints.createTarget)
```
Comment (agustito37) [inline src/components/targets/CreateTargetForm/index.js]:
```suggestion
  validate: validations(createTarget)
```

File: src/components/targets/CreateTargetForm/index.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { func, string, bool } from 'prop-types';
+import { Field, reduxForm } from 'redux-form/immutable';
+import { View, Text } from 'react-native';
+
+import * as constraints from 'utils/constraints';
```
Comment (agustito37) [inline src/components/targets/CreateTargetForm/index.js]:
```suggestion
import { validations, createTarget } from 'utils/constraints';
```

File: src/containers/MainScreen/index.js [hunk 6]
```diff
@@ -70,9 +106,15 @@ MainScreen.navigationOptions = {
 
 const mapState = state => ({
   username: getUser(state).username,
-  currentPosition: getCurrentPosition(state).currentPosition
+  currentPosition: getCurrentPosition(state).currentPosition,
```
Comment (agustito37) [inline src/containers/MainScreen/index.js]:
```suggestion
  currentPosition: getCurrentPosition(state),
```

File: src/containers/MainScreen/index.js [hunk 7]
```diff
@@ -70,9 +106,15 @@ MainScreen.navigationOptions = {
...
+  lastClickPosition: getLastClickPosition(state).lastClickPosition
```
Comment (agustito37) [inline src/containers/MainScreen/index.js]:
```suggestion
  lastClickPosition: getLastClickPosition(state)
```

File: src/selectors/mapSelector.js [hunk 8]
```diff
@@ -4,3 +4,8 @@ export const getCurrentPosition = createSelector(
   state => state.getIn(['map']),
   currentPosition => currentPosition.toJS()
 );
+
+export const getLastClickPosition = createSelector(
+  state => state.getIn(['map']),
```
Comment (agustito37) [inline src/selectors/mapSelector.js]:
```suggestion
  state => state.getIn(['map', 'lastClickPosition']),
```

File: src/selectors/mapSelector.js [hunk 9]
```diff
@@ -4,3 +4,8 @@ export const getCurrentPosition = createSelector(
...
+  lastClickPosition => lastClickPosition.toJS()
```
Comment (agustito37) [inline src/selectors/mapSelector.js]:
```suggestion
  lastClickPosition => lastClickPosition
```

**Pull request merged**